ListingManager/ListingManager.cs

```diff
@@ -0,0 +1,306 @@
+
+            string directoryName = directoryNameFull
+                .Substring(lastIndexOfPathSep, directoryNameFull.Length - lastIndexOfPathSep);
+
+            return fileNameRegex.IsMatch(path) && !directoryName.Contains(".Tests");
+        }
+
+        public static void UpdateListingNamespace(string path, int chapterNumber, string listingNumber,
+            string listingData, bool verbose = false, bool preview = false)
+        {
+            string paddedChapterNumber = new string(chapterNumber+"").PadLeft(2, '0');
+            string paddedListingNumber = new string(listingNumber).PadLeft(2, '0');
+            
+            string newFileNameTemplate = "Listing{0}.{1}{2}.cs";
+            string newNamespace = "AddisonWesley.Michaelis.EssentialCSharp" +
+                                  $".Chapter{paddedChapterNumber}" +
+                                  $".Listing{paddedChapterNumber}_" +
+                                  $"{paddedListingNumber}";
+            string newFileName = string.Format(newFileNameTemplate,
+                paddedChapterNumber,
+                paddedListingNumber,
+                string.IsNullOrWhiteSpace(listingData) ? "" : $".{listingData}");
+
+            if (verbose)
+            {
+                Console.WriteLine($"Corrective action. {Path.GetFileName(path)} rename to {newFileName}");
+            }
+            
+            if (!preview) UpdateNamespaceOfPath(path, newNamespace, newFileName);
+        }
+        
+        public static bool UpdateNamespaceOfPath(string path, string newNamespace, string newFileName = "")
+        {
+            if (!Path.HasExtension(path) || Path.GetExtension(path) != ".cs")
+            {
+                return false;
+            }
+
+            // read file into memory
+            string[] allLinesInFile = File.ReadAllLines(path);
+            
+            File.Delete(path);
+
+            string targetPath = Path.Combine(Path.GetDirectoryName(path), newFileName) ?? path;
+
+            using (TextWriter textWriter = new StreamWriter(targetPath, true))
+            {
+                foreach (string line in allLinesInFile)
+                {
+                    if (line.StartsWith("namespace"))
+                    {
+                        textWriter.WriteLine("namespace " + newNamespace);
+                    }
+                    else
+                    {
+                        textWriter.WriteLine(line);
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        public static bool GetPathToAccompanyingUnitTest(string listingPath, out string pathToTest)
+        {
+            string testDirectory = $"{Path.GetDirectoryName(listingPath)}.Tests";
+
+            Regex regex = new Regex(@"((Listing\d{2}.\d{2})([A-Z]?)((\.Tests)?)).*.cs$");
+
+            string testFileName = regex.IsMatch(listingPath) ? regex.Match(listingPath).Groups[1].Value : "";
```

Comment:
You probably just want to do the `.Match` call then check the `.Success` on the `Match` object